test_scripts/Defects/5_0/1006/008_SubscribeButton_UNSUPPORTED_RESOURCE_UI_isReady_false.lua

```diff
@@ -0,0 +1,40 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/1006
+---------------------------------------------------------------------------------------------------
+-- Description: SDL sends SubscribeButton response with UNSUPPORTED_RESOURCE, success:false for single UI-related RPC
+--  in case HMI respond UI.IsReady with "available" = false
+--
+-- Precondition:
+-- 1. SDL and HMI are started.
+-- 2. HMI respond with 'available' = false on UI.IsReady request from SDL
+-- 3. App is registered and activated
+-- Steps:
+-- 1. App requests SubscribeButton with UI part to SDL
+-- SDL does:
+-- - not send Buttons.SubscribeButton request to HMI
+-- - respond SubscribeButton(resultCode: UNSUPPORTED_RESOURCE, success: false) to App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require("test_scripts/Defects/5_0/1006/common")
+
+--[[ Local Variables ]]
+local rpcParams = {
+  rpc = "SubscribeButton",
+  requestParam = { buttonName = "OK" },
+  expectExtraRequest = function()
+    common.getHMIConnection():ExpectRequest("Buttons.SubscribeButton"):Times(0)
+  end
```

Comment:
Maybe ButtonPress also should be checked from Buttons interface

Reply:
will be covered with https://luxproject.luxoft.com/jira/browse/FORDTCN-13042